pkg/log/zap/flags.go

```diff
@@ -29,16 +29,15 @@ import (
 )
 
 var levelStrings = map[string]zapcore.Level{
-	"debug":  zap.DebugLevel,
-	"-1":     zap.DebugLevel,
-	"info":   zap.InfoLevel,
-	"0":      zap.InfoLevel,
-	"error":  zap.ErrorLevel,
-	"2":      zap.ErrorLevel,
-	"dpanic": zap.DPanicLevel,
-	"panic":  zap.PanicLevel,
-	"warn":   zap.WarnLevel,
-	"fatal":  zap.FatalLevel,
+	"debug": zap.DebugLevel,
+	"info":  zap.InfoLevel,
+	"error": zap.ErrorLevel,
+}
+
+var stackLevelStrings = map[string]zapcore.Level{
+	"disabled": zap.FatalLevel,
```

Comment:
If we're allowing all stacktraces to be completely disabled, we need to mention what `disabled` means in the `--zap-stacktrace-level` flag help text since it's not a zap level.
https://github.com/kubernetes-sigs/controller-runtime/blob/3156051da1ede28679182e1918b226c6f3fa8a74/pkg/log/zap/zap.go#L247-L248